src/os/bluestore/BlueStore.cc

```diff
@@ -7170,39 +7110,26 @@ void BlueStore::_txc_state_proc(TransContext *txc)
     case TransContext::STATE_KV_SUBMITTED:
       txc->log_state_latency(logger, l_bluestore_state_kv_committing_lat);
       txc->state = TransContext::STATE_KV_DONE;
-      _txc_finish_kv(txc);
+      _txc_committed_kv(txc);
```

Comment:
the flush() call currently just enforces correct ordering, e.g.,

 - omap_set a=b
 - omap_clear  (needs to see 'a' key in order to remove it)

it doesn't have anything to do with committed or durability; that's what the callbacks are for (so the caller can reason about that).